src/sortingtraining/SortTester.java

```diff
@@ -21,34 +18,25 @@ void run() {
 
 		try (Scanner in = new Scanner(System.in)) {
 
-			int[] numbers;
-
-			numbers = getInts(in);
-
-			mainMenu(in, numbers);
+			mainMenu(in);
 		}
 	}
 
-	private void mainMenu(Scanner in, int[] numbers) {
+	private void mainMenu(Scanner in) {
 
 		while (true) {
 
-			while (true) {
-				System.out.print(
-						"Type a sorting strategy (BUBBLE, BUCKET, INSERTION, QUICK, DEFAULT, ALL(compare all), EXIT): ");
-				try {
-					strategy = Strategy.valueOf(in.nextLine().toUpperCase());
-
-					break;
-				} catch (IllegalArgumentException e) {
-					System.out.println("Wrong input, try again!");
-				}
-			}
+			Strategy strategy = getStrategy(in);
 
 			if (strategy == Strategy.EXIT) {
+
+				System.out.println("\nSee you again!\n");
 				break;
+			}
+
+			int[] numbers = getInts(in);
 
-			} else if (strategy == Strategy.ALL) {
+			if (strategy == Strategy.COMPARE) {
```

Comment:
Have you tried if comparison `(null == Strategy.COMPARE)` would throw NPE? 
I know for 100% sure, that constant to anything (`null `including) is NPE safe.

Reply:
It is throwing it. I can do it by introducing local variable sortingStrategy and checking if sortingStrategy == null

So i left it as it is.